src/main-client.js

```diff
@@ -0,0 +1,120 @@
+import $ from 'jquery'
+import moment from 'moment'
+import Hotel from './Hotel';
+import Booking from './Booking';
+import Room from './Room';
+
+let url, urlString, userId, hotel
+
+function fetchUserData(id) {
+  return fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/users/users')
+    .then(response => response.json())
+    .then(data => data.users.find(user => user.id === Number(id)))
+    .catch(error => console.log(error))
+}
+
+function fetchBookingData() {
+  return fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings')
+    .then(response => response.json())
+    .then(data => data.bookings)
+    .catch(error => console.log(error))
+}
+
+function fetchRoomData() {
+  return fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/rooms/rooms')
+    .then(response => response.json())
+    .then(data => data.rooms)
+    .catch(error => console.log(error))
+}
+
+function getUserDashBoard(id) {
+  Promise.all([fetchRoomData(), fetchBookingData(), fetchUserData(id)])
+    .then(value => {
+      let user = value[2]
+      let bookings = value[1].map(booking => new Booking(booking))
+      let rooms = value[0].map(room => new Room(room))
+      hotel = new Hotel(rooms, bookings);
+      welcomeUser(user.name)
+      let userBookings = hotel.getUserBookings(user.id)
+      let todaysDate = moment().format('YYYY/MM/DD')
+      $('.booking-input-field').attr('min', todaysDate.split('/').join('-'))
+      getPastBookingsOnDom(userBookings, hotel.rooms, todaysDate)
+    })
+}
+
+if (window.location.href.includes("userId")) {
+  urlString = window.location.href
+  url = new URL(urlString)
+  userId = url.searchParams.get('userId')
+  getUserDashBoard(userId)
+  // $('.booking-date-button').on('click', filterRooms)
+}
+
+function welcomeUser(name) {
+  $('.client-welcome-heading').text(`Welcome ${name.split(' ')[0]}`)
+}
+
+function getPastBookingsOnDom(bookings, rooms, todaysDate) {
+  let upcomingBookings = bookings.filter(booking => booking.date >= todaysDate)
```

Comment:
As it pertains to bookings, might fit better as a method on the class.

Reply:
Same with calculateBookingCosts below

Reply:
I will definitely try to refactor this and add these functions to the appropriate classes.